client/scripts/views/modals/onboarding_modal/claim_with_cli.ts

```diff
@@ -0,0 +1,166 @@
+import 'modals/onboarding_modal/claim_with_cli.scss';
+
+import m, { Vnode } from 'mithril';
+import { Form, FormGroup, Input, Checkbox, Spinner } from 'construct-ui';
+import { Account, ChainBase } from 'models';
+import app from 'state';
+import { isU8a, isHex } from '@polkadot/util';
+import { createUserWithAddress } from 'controllers/app/login';
+import Substrate from 'controllers/chain/substrate/main';
+import CodeBlock from 'views/components/widgets/code_block';
+
+import { onboardingCWIcon, onboardingArrowLeftRightIcon, onboardingCLIIcon } from '../../components/sidebar/icons';
+import AddressSwapper from '../../components/addresses/address_swapper';
+
+import OnboardingFooterActions from './footer_actions';
+
+
+interface IOnboardingCLIAttr {
+  onNext: () => void;
+  onBack: () => void;
+  address: string;
+  accountVerifiedCallback: (account: Account<any>, onNext: (account: Account<any>) => void) => void;
+}
+
+const OnboardingCLI: m.Component<IOnboardingCLIAttr, {
+  error;
+  userProvidedSignature: string;
+  secretPhraseSaved: boolean;
+  newAddress: Account<any>;
+  isEd25519: boolean;
+  loadingNewAddress: boolean;
+}> = {
+  oninit: async (vnode) => {
+    const { address } = vnode.attrs;
+    vnode.state.error = null;
+    vnode.state.loadingNewAddress = true;
+    vnode.state.newAddress = null;
+
+    // Prevent validation on empty field
+    if (address === '') {
+      vnode.state.loadingNewAddress = false;
+      vnode.state.error = 'Address should not be empty.';
+      return;
+    }
+
+    if ((app.chain.base === ChainBase.Substrate)) {
```

Comment:
This block shouldn't be duplicated in multiple places. We should factor it into a utility.

Reply:
please help me for this one